README.md
[![NPM](https://nodei.co/npm/yocto-atos.png?downloads=true&downloadRank=true&stars=true)](https://nodei.co/npm/yocto-atos/)

![alt text](https://david-dm.org/yoctore/yocto-atos.svg "Dependencies Status")
[![Code Climate](https://codeclimate.com/github/yoctore/yocto-atos/badges/gpa.svg)](https://codeclimate.com/github/yoctore/yocto-atos)
[![Test Coverage](https://codeclimate.com/github/yoctore/yocto-atos/badges/coverage.svg)](https://codeclimate.com/github/yoctore/yocto-atos/coverage)
[![Issue Count](https://codeclimate.com/github/yoctore/yocto-atos/badges/issue_count.svg)](https://codeclimate.com/github/yoctore/yocto-atos)
[![Build Status](https://travis-ci.org/yoctore/yocto-atos.svg?branch=master)](https://travis-ci.org/yoctore/yocto-atos)

## Overview

This module is a part of yocto node modules for NodeJS.

Please see [our NPM repository](https://www.npmjs.com/~yocto) for complete list of available tools (completed day after day).

An utility tool to provide payment via Atos API

## Motivation

After each development, conclusion is the same : we need to create an utility tools with all our utility method to be able to reuse them in other program. That's why we create this utility tools.

*Although this module was completed day after day.*

## How to use
  - First you need to create an Store with Atos, and retrieve <client_id> and <secret_key>

All methods returned promise for success of fail

### Load config

This wrapper use the two mode of Atos : 'sandbox' and 'live'
The module must be loaded like this :

```javascript

var atos = require('yocto-atos')();

atos.loadConfig({
  // "sandbox" or "live"
  mode : 'sandbox',
  secretKey         : '484GEF87A45AEAA87842634A684A354A643541AA',
  config : {
    currencyCode      : '978',
    keyVersion        : 1,
  }
}).then(function (config) {
  console.log('success')
}).catch(function (error) {
  console.log('error : ', error)
});
```

### Create authorization payment

An Atos authorization payment permit you to capture payment during in period of 29 days.
*NB :* Atos will honor the funds for a 6-day period after the basic authorization

#### With credit card

To create payment with an credit card

```javascript
var paymentData = {
  amount                : 123,
  cardNumber            : '559955995599559955',
  cardExpiryDate        : '201905',
  cardCSCValue          : '985',
  merchantId            : '1',
  orderId               : '57358b7fea9d4398641209e5', // optional
  transactionReference  : '57358b7fea9d4398641209e5', // optional
  s10TransactionReference : { // optional
    s10TransactionId      : '777888899445566112233',
    s10TransactionIdDate  : '20050606'
  }
};

atos.modules.creditCard.createAuthorization(paymentData).then(function (config) {
  console.log('success')
}).catch(function (error) {
  console.log('error : ', error)
});
```

### Capture payment

TO capture an payment you should have an valid capture payment. The field "transactionReference" or "s10TransactionReference" should be set.

```javascript
var captureData = {
  operationAmount       : 100,
  merchantId            : '1',
  transactionReference  : '57358b7fea9d4398641209e5',
  s10TransactionReference : {
    s10TransactionId      : '777888899445566112233',
    s10TransactionIdDate  : '20050606'
  }
};

atos.capturePayment(captureData).then(function (config) {
  console.log('success')
}).catch(function (error) {
  console.log('error : ', error)
});
```

### Cancel payment

An payment authorization can be void only if was previously captured and not sent to the bank. The field "transactionReference" or "s10TransactionReference" should be set.

```javascript

var cancelData = {
  transactionReference  : '3aabcedaapfa1f33a00425aa8',
  operationOrigin       : 'my app',
  merchantId            : '1',
  operationAmount       : 974
  s10TransactionReference : {
    s10TransactionId      : '777888899445566112233',
    s10TransactionIdDate  : '20050606'
  }
}

atos.cancelPayment(cancelData).then(function () {
  console.log('success')
}).catch(function (error) {
  console.log('error : ', error)
});
```


CHANGELOG.md
## 1.1.0 (2016-06-29)
- Change init object by passing now an array of shops to handle multiple shop config

## 1.0.1 (2016-06-29)
- Allow more Data into create card authorization
- Update seal calcul method

## 1.0.0 (2016-06-28)
- Allow more Data into create card authorization

## 0.1.0 (2016-06-27)
- Initialize npm module


test/unit/index.js
var utils  = require('yocto-utils');
var chai   = require('chai').assert;
var expect = require('chai').expect;
var should = require('chai').should;
var _      = require('lodash');
var atos = require('../../src/')();
var util   = require('util');

atos.logger.enableConsole(false);

var types = [ null, undefined, 1, true, false, NaN, 'a', '', {}, [] ];

var config = {
  mode      : 'sandbox',
  shops : [
    {
      label : 'test',
      merchantId : '1110',
      secretKey : 111,
      config    : {
        currencyCode : '978',
        keyVersion   : 1
      }
    }
  ]
};

describe('Load config', function () {

  // valid conf
  describe('loadConfig() must return a return success promise : ', function () {
    // test load atos with promised
    it('Use valid configuration', function( ) { // no done

      // note the return
      return atos.loadConfig(config).then(function( data){
        expect(data).to.equal(config);
      });// no catch, it'll figure it out since the promise is rejected
    });
  });

  // wrong conf
  describe('loadConfig() must return a return fail promise because < mode > will not be an ' +
  'valid value : ', function () {

    types.forEach(function (t) {
      it('Using type for : < ' +  util.inspect(t, { depth : null }) + ' > for field < mode >',
      function () {

        var config = {
          mode          : t,
          shops : [
            {
              label : 'test',
              merchantId : '1110',
              secretKey : 111,
              config    : {
                currencyCode : '978',
                keyVersion   : 1
              }
            }
          ]
        };
        // note the return
        return atos.loadConfig(config).catch(function (data) {
          expect(data).to.be.an('string');
        });// no catch, it'll figure it out since the promise is rejected
      });
    });
  });

  // wrong conf
  describe('loadConfig() must return a return fail promise because < mode > is valid but ' +
  'other value not : ', function () {

    // overide types by removing string value
    var types = [ null, undefined, 1, true, false, NaN, '', {}, [] ];

    types.forEach(function (t) {
      it('Using type for : < ' +  util.inspect(t, { depth : null }) + ' > for field < mode >',
      function () {

        var config = {
          mode          : 'sandbox',
          shops : [
            {
              label : t,
              merchantId : t,
              secretKey : t,
              config    : {
                currencyCode : t,
                keyVersion   : t
              }
            }
          ]
        };
        // note the return
        return atos.loadConfig(config).catch(function (data) {
          expect(data).to.be.an('string');
        });// no catch, it'll figure it out since the promise is rejected
      });
    });
  });
});


Gruntfile.js
'use strict';

module.exports = function (grunt) {
  // init config
  grunt.initConfig({
    // default package
    pkg       : grunt.file.readJSON('package.json'),

    todo      : {
      options : {
        marks       : [
          { name : 'TODO', pattern : /TODO/, color : 'yellow' },
          { name : 'FIXME', pattern : /FIXME/, color : 'red' },
          { name : 'NOTE', pattern : /NOTE/, color : 'blue' }
        ],
        file        : 'REPORT.md',
        githubBoxes : true,
        colophon    : true,
        usePackage  : true
      },
      src     : [
        'src/validators/**/*.js',
        'src/methods/**/*.js'
      ]
    },

    // hint our app
    yoctohint : {
      options  : {},
      all      : [ 'src/***', 'Gruntfile.js' ]
    },

    // Uglify our app
    uglify    : {
      options : {
        banner  : '/* <%= pkg.name %> - <%= pkg.description %> - V<%= pkg.version %> */\n'
      },
      api     : {
        files : [ {
          expand  : true,
          cwd     : 'src',
          src     : '**/*.js',
          dest    : 'dist'
        } ]
      }
    },
    clean     : {
      dist        : [ 'dist/*']
    },
    // test our app
    mochacli  : {
      options : {
        'reporter'       : 'spec',
        'inline-diffs'   : false,
        'no-exit'        : true,
        'force'          : false,
        'check-leaks'    : true,
        'bail'           : false
      },
      all     : [ 'tests/unit/*.js' ]
    },
    // copy files
    copy      : {
      all : {
        expand  : true,
        cwd     : 'src/',
        src     : '**',
        dest    : 'dist/'
      }
    },
    // unit testing
    mochaTest : {
      // Test all unit test
      all  : {
        options : {
          reporter : 'spec',
        },
        src     : [ 'test/unit/*.js' ]
      }
    }
  });

  // Load the plugins
  grunt.loadNpmTasks('grunt-contrib-uglify');
  grunt.loadNpmTasks('grunt-mocha-test');
  grunt.loadNpmTasks('yocto-hint');

  grunt.registerTask('hint', 'yoctohint');
  grunt.registerTask('test', 'mochaTest');
  grunt.registerTask('build', [ 'hint', 'test', 'uglify' ]);
  grunt.registerTask('default', 'build');
};


example/crypto.js
var utils = require('../src/modules/utils/')();

var paymentData = {
  amount                : '974',
  cardNumber            : '5017670000008300',
  cardExpiryDate        : '201705',
  cardCSCValue          : '985',
  orderId               : '575acedaa8fa1f33004254a1',
  transactionReference  : '82abc2paapfa1f33a00425aa8',
  merchantId            : '201000007070001',
  customerIpAddress     : '192.168.2.101',
  invoiceReference      : '8216062410201313',
  //customerId            : '574eb5e8b7e67f7832f562fc',
  billingContact        : {
    firstname : 'foo',
    lastname  : 'bar'
  },
  billingAddress        : {
    country   : 'REU',
    addressAdditional1 : 'toto route',
    zipCode : '97400'
  },
  customerContact        : {
    firstname : 'foo',
    lastname  : 'bar'
  },
  customerAddress        : {
    country   : 'REU',
    addressAdditional1 : 'toto route',
    zipCode : '97400'
  },
  shoppingCartDetail  : {
    shoppingCartTotalTaxAmount : 50,
    shoppingCartItemList : [{
      productSKU  : '97434567891234',
      productCode : '1234567891234',
      productName : 'toto'
    }]
  }
  // fraudData             : {
  //   allowedCardCountryList  : [ 'FRA', 'MUS', 'REU', 'MDG' ],
  //   allowedIpCountryList    : [ 'FRA', 'MUS', 'REU', 'MDG' ]
  // }
};

var seal = utils.calculSEAL(paymentData, 'AJEF262FAFE51562EAF6EAF1421SA6');
console.log(' \n ===> SEAL is : ', seal);


src/modules/creditCard/index.js
'use strict';

var _       = require('lodash');
var logger  = require('yocto-logger');
var joi     = require('joi');
var Q       = require('q');
var utils   = require('yocto-utils');

/**
* Yocto Atos : Atos wrapper for Sips Office Json Interface
*
* @date   : 06/06/2016
* @author : Cedric Balard <cedric@yocto.re>
* @copyright : Yocto SAS, All right reserved
* @class CreditCard
*/
function CreditCard (yLogger, config) {

  // config
  this.config = config || {};

  // default logger
  this.logger = yLogger;

  // require api
  this.api = require('../api')(logger);
}

/**
 * Method that trying to retrieve authorization for an credit card to make payment
 *
 * @param  {Object} paymentData the necessary data to create payment authorization
 * @return {boolean} true if file is loaded, otherwise false
 */
CreditCard.prototype.createAuthorization = function (paymentData) {
  // create a promise deferred
  var deferred = Q.defer();

  // Joi schema for create authorization
  var schema = joi.object().required().keys({
    amount                  : joi.number().integer().required().min(0),
    captureDay              : joi.number().integer().optional().min(0).default(6),
    orderChannel            : joi.string().optional().default('INTERNET'),
    captureMode             : joi.string().optional().default('VALIDATION'),
    paymentPattern          : joi.string().optional().default('ONE_SHOT'),
    cardNumber              : joi.string().creditCard().required(),
    cardExpiryDate          : joi.string().required().empty(),
    cardCSCValue            : joi.string().required().empty(),
    orderId                 : joi.string().optional().empty(),
    transactionReference    : joi.string().optional().empty(),
    interfaceVersion        : joi.string().optional().default('IR_WS_2.12'),
    // Identifier of the shop, this value is provided to the merchant by Sips
    merchantId              : joi.string().optional().empty(),
    customerId              : joi.string().optional().empty(),
    customerIpAddress       : joi.string().optional().empty(),
    invoiceReference        : joi.string().optional().empty(),
    transactionOrigin       : joi.string().optional().empty(),
    billingAddress          : joi.object().optional().keys({
      addressAdditional1  : joi.string().optional().empty(),
      addressAdditional2  : joi.string().optional().empty(),
      addressAdditional3  : joi.string().optional().empty(),
      city                : joi.string().optional().empty(),
      company             : joi.string().optional().empty(),
      country             : joi.string().optional().length(3),
      postBox             : joi.string().optional().empty(),
      state               : joi.string().optional().empty(),
      street              : joi.string().optional().empty(),
      streetNumber        : joi.string().optional().empty(),
      zipCode             : joi.string().optional().empty(),
    }),
    billingContact          : joi.object().optional().keys({
      email     : joi.string().email().optional().empty(),
      firstname : joi.string().optional().empty(),
      gender    : joi.string().optional().empty().allow([ 'M', 'F']),
      lastname  : joi.string().optional().empty(),
      mobile    : joi.string().optional().empty(),
      phone     : joi.string().optional().empty(),
      title     : joi.string().optional().empty()
    }),
    customerAddress         : joi.object().optional().keys({
      addressAdditional1  : joi.string().optional().empty(),
      addressAdditional2  : joi.string().optional().empty(),
      addressAdditional3  : joi.string().optional().empty(),
      city                : joi.string().optional().empty(),
      company             : joi.string().optional().empty(),
      country             : joi.string().optional().length(3),
      postBox             : joi.string().optional().empty(),
      state               : joi.string().optional().empty(),
      street              : joi.string().optional().empty(),
      streetNumber        : joi.string().optional().empty(),
      zipCode             : joi.string().optional().empty(),
    }),
    customerContact         : joi.object().optional().keys({
      email     : joi.string().email().optional().empty(),
      firstname : joi.string().optional().empty(),
      gender    : joi.string().optional().empty().allow([ 'M', 'F']),
      lastname  : joi.string().optional().empty(),
      mobile    : joi.string().optional().empty(),
      phone     : joi.string().optional().empty(),
      title     : joi.string().optional().empty()
    }),
    deliveryAddress         : joi.object().optional().keys({
      addressAdditional1  : joi.string().optional().empty(),
      addressAdditional2  : joi.string().optional().empty(),
      addressAdditional3  : joi.string().optional().empty(),
      city                : joi.string().optional().empty(),
      company             : joi.string().optional().empty(),
      country             : joi.string().optional().length(3),
      postBox             : joi.string().optional().empty(),
      state               : joi.string().optional().empty(),
      street              : joi.string().optional().empty(),
      streetNumber        : joi.string().optional().empty(),
      zipCode             : joi.string().optional().empty(),
    }),
    deliveryContact         : joi.object().optional().keys({
      email     : joi.string().email().optional().empty(),
      firstname : joi.string().optional().empty(),
      gender    : joi.string().optional().empty().allow([ 'M', 'F']),
      lastname  : joi.string().optional().empty(),
      mobile    : joi.string().optional().empty(),
      phone     : joi.string().optional().empty(),
      title     : joi.string().optional().empty()
    }),
    shoppingCartDetail      : joi.object().optional().keys({
      shoppingCartTotalAmount     : joi.number().integer().optional().min(0),
      shoppingCartTotalQuantity   : joi.number().integer().optional().min(0),
      shoppingCartTotalTaxAmount  : joi.number().integer().optional().min(0),
      // The most expensive product in the shopping cart.
      mainProduct                 : joi.string().optional().empty(),
      shoppingCartItemList        : joi.array().items(
        joi.object().keys({
          // Monetary value of the tax for the product (unit)
          productUnitTaxAmount  : joi.number().optional().min(0),
          productName           : joi.string().optional().empty(),
          productDescription    : joi.string().optional().empty(),
          // Code of the ordered product.
          productCode           : joi.string().optional().empty(),
          // Merchant’s product identifier code, sent back in the response without modification.
          productSKU            : joi.string().optional().empty(),
          // Unit amount of the product
          productUnitAmount     : joi.number().optional().min(0),
          productQuantity       : joi.number().optional().min(0),
          productTaxRate        : joi.number().optional().min(0),
          productCategory       : joi.string().optional().empty()
        })
      ).optional()
    }),
    fraudData               : joi.object().optional().keys({
      allowedCardCountryList  : joi.array().optional().items(
        joi.string().length(3)
      ),
      allowedIpCountryList    :  joi.array().optional().items(
        joi.string().length(3)
      ),
      bypassCtrlList          : joi.array().optional().items(
        joi.string().empty()
      ),
      bypassInfoList          : joi.string().optional().empty().allow([
        'IpCountry', 'Card', 'All'
      ]),
      deniedCardCountryList   :  joi.array().optional().items(
        joi.string().length(3)
      ),
      deniedIpCountryList     :  joi.array().optional().items(
        joi.string().length(3)
      )
    })
  });

  // validate joi schema with the given file
  var result   = schema.validate(paymentData);

  // check if an error occured
  if (result.error) {
    // An error occured joi schema is not conform
    this.logger.error('[ YoctoAtos.CreditCard.createAuthorization.joi ] - an error' +
    ' occured schema is not conform, more details : ' + utils.obj.inspect(result.error));
    // Config file was not loaded
    deferred.reject(result.error.toString());
    return deferred.promise;
  }

  // override var with default value
  paymentData = result.value;

  this.logger.debug('[ YoctoAtos.CreditCard.createAuthorization ] - a new request will' +
  ' be send to create authorization payment');

  // Call api module to make an request to atos
  this.api.process(this.config, 'checkout/cardOrder', paymentData, 'createAuthorization', false).
  then(function (value) {

    // resolve success
    deferred.resolve(value);
  }).catch(function (error) {

    this.logger.error('[ YoctoAtos.CreditCard.createAuthorization.create ] - the' +
    ' payment was not authorized');

    // reject error
    deferred.reject(error);
  }.bind(this));

  // return promise of process
  return deferred.promise;
};

/**
 * Method that capture an existing authorization pauymebt
 *
 * @param  {Object} captureData the necessary data to capture authorization
 * @return {boolean} true if file is loaded, otherwise false
 */
CreditCard.prototype.capturePayment = function (captureData) {
  // create a promise deferred
  var deferred = Q.defer();

  // Joi schema for create authorization
  var schema = joi.object().required().keys({
    operationAmount         : joi.number().integer().required().min(0),
    transactionReference    : joi.string().optional().empty(),
    // Identifier of the shop, this value is provided to the merchant by Sips
    merchantId              : joi.string().required().empty(),
    interfaceVersion        : joi.string().optional().default('CR_WS_2.12'),
    operationOrigin         : joi.string().optional().empty()
  });

  // validate joi schema with the given file
  var result   = schema.validate(captureData);

  // check if an error occured
  if (result.error) {
    // An error occured joi schema is not conform
    this.logger.error('[ YoctoAtos.CreditCard.capturePayment.joi ] - an error' +
    ' occured schema is not conform, more details : ' + result.error.toString());
    // Config file was not loaded
    deferred.reject(result.error.toString());
    return deferred.promise;
  }

  // override var with default value
  captureData = result.value;

  this.logger.debug('[ YoctoAtos.CreditCard.capturePayment ] - a new request will' +
  ' be send to create authorization payment');

  // Call api module to make an request to atos to capture payment
  this.api.process(this.config, 'cashManagement/validate', captureData, 'capturePayment').then(
  function (value) {

    // resolve success
    deferred.resolve(value);
  }).catch(function (error) {

    this.logger.error('[ YoctoAtos.CreditCard.capturePayment ] - the' +
    ' payment was not authorized');

    // reject error
    deferred.reject(error);
  }.bind(this));

  // return promise of process
  return deferred.promise;
};

/**
 * Method that cancel an payment cancel
 *
 * @param  {Object} cancelData the necessary data to cancel payment authorization
 * @return {boolean} true if file is loaded, otherwise false
 */
CreditCard.prototype.cancelPayment = function (cancelData) {
  // create a promise deferred
  var deferred = Q.defer();

  // Joi schema for create authorization
  var schema = joi.object().required().keys({
    operationAmount         : joi.number().integer().required().min(0),
    s10TransactionReference : joi.object().optional().keys({
      s10TransactionId     : joi.string().required().empty(),
      s10TransactionIdDate : joi.string().required().empty(),
    }),
    // Identifier of the shop, this value is provided to the merchant by Sips
    merchantId              : joi.string().required().empty(),
    interfaceVersion        : joi.string().optional().default('CR_WS_2.12'),
    transactionReference    : joi.string().optional().empty(),
    operationOrigin         : joi.string().optional().empty()
  });

  // validate joi schema with the given file
  var result   = schema.validate(cancelData);

  // check if an error occured
  if (result.error) {
    // An error occured joi schema is not conform
    this.logger.error('[ YoctoAtos.CreditCard.cancelPayment.joi ] - an error' +
    ' occured schema is not conform, more details : ' + result.error.toString());
    // Config file was not loaded
    deferred.reject(result.error.toString());
    return deferred.promise;
  }

  // override var with default value
  cancelData = result.value;

  this.logger.debug('[ YoctoAtos.CreditCard.cancelPayment ] - a new request will' +
  ' be send to create authorization payment');

  // Call api module to make an request to atos to capture payment
  this.api.process(this.config, 'cashManagement/cancel', cancelData, 'capturePayment').then(
  function (value) {

    // resolve success
    deferred.resolve(value);
  }).catch(function (error) {

    this.logger.error('[ YoctoAtos.CreditCard.cancelPayment ] - the' +
    ' payment was not authorized');

    // reject error
    deferred.reject(error);
  }.bind(this));

  // return promise of process
  return deferred.promise;
};

// Default export
module.exports = function (l, config) {
  // is a valid logger ?
  if (_.isUndefined(l) || _.isNull(l)) {
    logger.warning('[ CreditCard.constructor ] - Invalid logger given. Use internal logger');
    // assign
    l = logger;
  }

  // default statement
  return new (CreditCard)(l, config);
};


example/creditCard.js
var atos = require('../src')();

var config = {
  mode      : 'sandbox',
  shops : [
    {
      label         : '#1',
      secretKey     : 'key1',
      merchantId    : '1',
      config : {
        currencyCode  : '978',
        keyVersion    : 1
      }
    },
    {
      label         : '#2',
      secretKey     : 'key 2',
      merchantId    : '2',
      config : {
        currencyCode  : '978',
        keyVersion    : 1
      }
    }
  ]
};

var paymentData = {
  amount                : '974',
  cardNumber            : '5017670000008300',
  cardExpiryDate        : '201705',
  cardCSCValue          : '985',
  orderId               : '575acedaa8fa1f33004254a1',
  transactionReference  : '97abc2paapfa1f33a00425aa8',
  merchantId            : '1',
  customerIpAddress     : '192.168.2.101',
  invoiceReference      : '8216062410201313',
  //customerId            : '574eb5e8b7e67f7832f562fc',
  billingContact        : {
    firstname : 'foo',
    lastname  : 'bar'
  },
  billingAddress        : {
    country   : 'REU',
    addressAdditional1 : 'toto route',
    zipCode : '97400'
  },
  customerContact        : {
    firstname : 'foo',
    lastname  : 'bar'
  },
  customerAddress        : {
    country   : 'REU',
    addressAdditional1 : 'toto route',
    zipCode : '97400'
  },
  shoppingCartDetail  : {
    shoppingCartTotalTaxAmount : 50,
    shoppingCartItemList : [
      {
        productSKU  : '1234567891234',
        productCode : '1234567891234',
        productName : 'toto'
      },
      {
        productSKU  : '8524567891234',
        productCode : '8524567891234',
        productName : 'tata'
      },
      {
        productSKU  : '9658467891234',
        productCode : '9658467891234',
        productName : 'titi'
      }
  ]
  }
  // fraudData             : {
  //   allowedCardCountryList  : [ 'FRA', 'MUS', 'REU', 'MDG' ],
  //   allowedIpCountryList    : [ 'FRA', 'MUS', 'REU', 'MDG' ]
  // }
};

var captureData = {
  operationAmount       : 974,
  transactionReference  : '97abc2paapfa1f33a00425aa8',
  merchantId            : '1',
  operationOrigin       : 'my app',
};

var cancelData = {
  transactionReference  : '3aabcedaapfa1f33a00425aa8',
  operationOrigin       : 'my app',
  merchantId            : '1',
  operationAmount       : 974
};

// // Create auth payment
// atos.loadConfig(config).then(function (value) {
//   console.log('===> success load, value is = ')
//
//   atos.modules.creditCard.createAuthorization(paymentData).then(function (value) {
//     console.log('\n===> success create auth payment, value is = ', value)
//   }).catch(function (error) {
//     console.log('\n===> error, is = ', error)
//   });
// }).catch(function (error) {
//   console.log('===> error, is = ', error);
// });

// CAptue paument
atos.loadConfig(config).then(function (value) {
  console.log('===> success load, value is = ')

  atos.modules.creditCard.capturePayment(captureData).then(function (value) {
    console.log('===> success create auth payment, value is = ', value)
  }).catch(function (error) {
    console.log('===> error, is = ', error)
  });
}).catch(function (error) {
  console.log('===> error, is = ', error);
});

// Cancel payment
// atos.loadConfig(config).then(function (value) {
//   console.log('===> success load, value is = ')
//
//   atos.modules.creditCard.cancelPayment(cancelData).then(function (value) {
//     console.log('===> success create auth payment, value is = ', value)
//   }).catch(function (error) {
//     console.log('===> error, is = ', error)
//   });
// }).catch(function (error) {
//   console.log('===> error, is = ', error);
// });
